prog/mutation.go

```diff
@@ -320,7 +320,28 @@ func (t *IntType) mutate(r *randGen, s *state, arg Arg, ctx ArgCtx) (calls []*Ca
 }
 
 func (t *FlagsType) mutate(r *randGen, s *state, arg Arg, ctx ArgCtx) (calls []*Call, retry, preserve bool) {
-	return mutateInt(r, s, arg)
+	if !t.BitMask {
+		return mutateInt(r, s, arg)
+	}
+
+	a := arg.(*ConstArg)
+	switch {
+	case r.nOutOf(1, 10):
+		a.Val = 0
+		for _, val := range t.Vals {
+			if r.bin() {
+				a.Val |= val
+			}
+		}
+	case r.nOutOf(2, 10):
+		a.Val = 0
```

Comment:
I think there is a high chance to produce the existing value (effectively no mutation). Consider that the current value is already 0.
The same can happen in the other cases of the switch if flags have few values (some flags have just 1 value).
We could have an outer loop that will retry the mutation until it produces a different value. It would ensure that we don't waste a test iteration by running the same program again.